server/controllers/profile.js

```diff
@@ -0,0 +1,123 @@
+const Profile = require("../models/profile");
+const asyncHandler = require("express-async-handler");
+const profile = require("../models/profile");
+
+// @route POST /profile/create
+// @desc create new Profile
+exports.createProfile=asyncHandler(async(req,res,next)=>{
+    const { user } = req.user.id;
+
+    const profileExists = await profile.findOne({ user });
+  
+    if (profileExists) {
+      res.status(400);
+      throw new Error("A user with that profile already exists");
+    }
+    const {
+        firstName,
+        lastName,
+        description,
+        phoneNo,
+        photoUrl,
+        email,
+        city,
+        DateOfBirth,
+        avaliabilty
+    }=req.body;
```

Comment:
Optional: We can validate the request body params and return a `400` if they're missing (You can do so within this controller. However, a more ideal way would be having a middleware applied to this route using `express-validator`)

Ditto where applicable

Reply:
To check the req.body params created a new validator and included in the router